packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -627,73 +641,43 @@ contract Bridge is EssentialContract, IBridge {
+        if (messageStatus[_msgHash] != _expectedStatus) revert B_INVALID_STATUS();
+    }
+
     ///@dev This implementation was suggested by OpenZeppelin in an audit to replace the previous.
-    function _hasEnoughGas(
+    function _hasInsufficientGas(
         uint256 _minGas,
         uint256 _dataLength
     )
-        internal
+        private
         view
         returns (bool result_)
     {
-        // The goal of the computation is to ensure that EIP-150 does not silently cap the amount of
-        // gas sent with the external call to be less than A. We note:
-        //
-        // - memory_cost: the amount of gas needed to expand the memory when storing the inputs and
-        //                outputs of the external call.
-        // - access_gas_cost: the gas cost of accessing the message.to account. This currently
-        //                    corresponds to 2600 gas if the account is cold, and 100 otherwise.
-        // - transfer_gas_cost: the cost of transferring a nonzero msg.value. This cost is currently
-        //                      9000, but provides a 2300 gas stipend to the called contract.
-        // - create_gas_cost: the cost of creating a new account, currently 25000. This only applies
-        //                    if message.value != 0, message.to.nounce == 0, message.to.code == b""
-        //                    and message.to.balance == 0. Because message.to is checked to have
-        // code,
-        //                    this cost can be  ignored here.
-
-        // We thus want to check that the following:
-        // 63 / 64 * (gasleft() - memory_cost - access_gas_cost - transfer_gas_cost) is at least as
-        // much as A (reference implementation). The sum of access_gas_cost and transfer_gas_cost
-        // can be upper bounded with 2_600 + 9_000 - 2_300 = 9_300. The memory_cost is cumbersome to
-        // compute in practice, but by estimating the costs through tests, we can upper bound the
-        // cost of memory expansion for up to 10_000 bytes of message.data in the context of the
-        // call with the formula 1_200 + 3 * message.data.length / 32.
-        //
-        // Thus, it would be possible to validate that the call will have enough gas by checking the
-        // following condition right before the external call is made:
-
         unchecked {
-            result_ = 63 * gasleft() >= 64 * _minGas + 63 * (10_500 + 3 * _dataLength / 32);
-        }
-
-        // The previous implementation is  `return gasleft() * 63) >> 6 >= gasLimit_`.
-    }
-
-    function _checkStatus(bytes32 _msgHash, Status _expectedStatus) private view {
-        if (messageStatus[_msgHash] != _expectedStatus) revert B_INVALID_STATUS();
-    }
+            // https://github.com/ethereum/execution-specs/blob/master/src/ethereum/cancun/vm/gas.py#L128
+            uint256 words = _dataLength / 32 + 1;
+            uint256 memoryGasCost = words * 3 + words * words / 512;
```

Comment:
I don't think we need to add this check here because we don't know if the memory expansion is required for this data inside the invocation call, and it's fully controlled by the user so nothing can be manipulated.

Also the `words * words / 512` part at least should be removed because that part should be done on the total memory size when it's done which is unknown here, so can't calculate it at this point.

Reply:
Sorry @Brechtpd , could you move this comment to https://github.com/taikoxyz/taiko-mono/pull/17422. I'm closing this issue in favor of another one.